packages/ui/src/components/Schemas/TextSchema.tsx

```diff
@@ -17,37 +17,39 @@ import { FontContext } from '../../contexts';
 type Props = SchemaUIProps & { schema: TextSchema };
 
 const TextSchemaUI = (
-  { schema, editable, placeholder, tabIndex, onChange }: Props,
+  { schema, editable, placeholder, tabIndex, onChange, onDynamicFontResize }: Props,
   ref: Ref<HTMLTextAreaElement>
 ) => {
   const font = useContext(FontContext);
-  const [dynamicFontSize, setDynamicFontSize] = useState<number | undefined>(undefined);
   const [fontAlignmentValue, setFontAlignmentValue] = useState<number>(0);
 
+  useEffect(() => {
+    getFontKitFont(schema, font).then((fontKitFont) => {
+      const fav = getFontAlignmentValue(fontKitFont, schema.fontSize ?? DEFAULT_FONT_SIZE);
+      setFontAlignmentValue(fav);
+    });
+  }, [schema, schema.width, font]);
 
   useEffect(() => {
```

Comment:
the `useEffect` was split into 2 because listening on a superset of attributes led the dynamic font sizing code into an infinite loop (or at least an extra execution). 
This is because we're adjusting the `schema.fontSize` value so we need to avoid listening for this change or we end up looping.